test/system/030-run.bats

```diff
@@ -1104,5 +1104,15 @@ EOF
     rm -rf $romount
 }
 
+@test "podman run --restart=always -- wait" {
+    # regression test for #18572 to make sure Podman waits less than 20 seconds
+    stdout=$PODMAN_TMPDIR/stdout
+    ctr=$(random_string)
+    run_podman run -d --restart=always --name=$ctr $IMAGE false
+    timeout --foreground -v --kill=10 20 \
+        $PODMAN wait $ctr > $stdout
+    is "$(< $stdout)" "1" "container should exit 1"
```

Comment:
You can just use `$PODMAN_TIMEOUT`, right now your test is missing the exit code check
e.g. https://github.com/containers/podman/blob/e2d1ad1fb2f59d7fe9685485bb69f6ff7987d74f/test/system/035-logs.bats#L329